lib/exercism/github.rb

```diff
@@ -21,18 +21,23 @@ def self.authenticate(code, client_id, client_secret)
       req.headers['User-Agent'] = user_agent
       req.body = { client_id: client_id, client_secret: client_secret, code: code }.to_json
     end
-    access_token = JSON.parse(response.body)['access_token']
+    JSON.parse(response.body)['access_token']
+  end
 
+  def self.user_info(access_token)
     conn = connect_to 'https://api.github.com'
     response = conn.get do |req|
       req.url '/user'
       req.headers['User-Agent'] = user_agent
       req.params['access_token'] = access_token
     end
-    result = JSON.parse(response.body)
-    [result['id'], result['login'], result['email'], result['avatar_url']]
+    if response.status != 200
+      raise UnauthorizedUser
+    else
+      result = JSON.parse(response.body)
+      [result['id'], result['login'], result['email'], result['avatar_url']]
```

Comment:
Thanks for clarifying, I see what you mean. Right now, `Github` is acting more like a module, since it is never instantiated and holds no data. You're right. Instead, it could be renamed to *be* the `GithubUser` or `GithubUserAuth` class and used in the same way.